### PR TITLE
Provide prebuilt std for gnullvm targets

### PR DESCRIPTION
**Pull request by mati865**
Revival of https://github.com/rust-lang/rust/pull/114346 which waiting on MCP that was accepted recently: https://github.com/rust-lang/compiler-team/issues/710#issuecomment-1942014308

### CODE REVIEW

**Comment by ehuss** [comment]
I'm not sure who is best to review this.

r? infra-ci

**Comment by Mark-Simulacrum** [comment]
Why is this not updating the platform support docs to indicate this is now a tier 2 target (without host tools)?

**Comment by mati865** [comment]
Fixed.

@rustbot ready

File: src/doc/rustc/src/platform-support/pc-windows-gnullvm.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # \*-pc-windows-gnullvm
 
-**Tier: 3**
+**Tier: 2 (without host tools)**
```
Comment (Mark-Simulacrum) [inline src/doc/rustc/src/platform-support/pc-windows-gnullvm.md:3]: Doesn't this need updating in https://doc.rust-lang.org/nightly/rustc/platform-support.html? Or are we auto-generating that?

**Comment by mati865** [comment]
Sorry, I was under impression this table is generated from other files.
Updated it and grepped through the repository.

Also found out the machine where I had built Dockerfile had not fully rebased branch and with current master this fails to build.
Marking as blocked on https://github.com/rust-lang/compiler-builtins/pull/575 which is blocked on https://github.com/rust-lang/rust/issues/121552

@rustbot blocked

**Comment by jeremyd2019** [comment]
rust-lang/compiler-builtins#575 is merged now, but that wound up breaking i686-pc-windows-gnu for me so it may still need more work, but it did fix the issue on i686-pc-windows-gnullvm

**Comment by mati865** [comment]
Yeah, breaking i686-pc-windows-gnu is unacceptable, so this PR remains blocked.

**Comment by jeremyd2019** [comment]
> I created [rust-lang/compiler-builtins#585](https://github.com/rust-lang/compiler-builtins/issues/585) to better track this (as opposed to a comment on the PR)

A fix for this has now been merged, and it looks like they even released a new version of the compiler-builtins crate (0.1.109)

**Comment by mati865** [comment]
Yeah, I already saw https://github.com/rust-lang/rust/pull/123719 and I'm waiting for it to merge.

**Comment by mati865** [comment]
Rebased, updated the toolchain and tested locally using `./src/ci/docker/run.sh dist-various-1`.

@rustbot ready

**Pull request merged**